packages/data-driver-postgres/src/utils/replace-markers.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Replace all `?` markers in a string with Postgres parameter markers (`$1`, `$2`, ...`$n`)
+ * @param sql - Any SQL string that contains markers that should be replaced
+ * @returns The final SQL query with appropriate parameters
+ */
+export const replaceMarkers = (sql: string) => {
+	let count = 0;
+	return sql.replace(/\?/g, () => `$${++count}`).trimEnd();
+};
```

Comment:
Doing direct string manipulations on the generated SQL can be very dangerous. 

Maybe we can use something like this (or alternatively use a generator function):

```ts
function integerGeneratorFactory(start: number) {
  let counter = start;

  return () => {
    const currentCount = counter;

    counter++;

    return currentCount;
  }
}
```

and call this in `constructSql` to initialize the generator and then pass it to each function that constructs a part of the SQL.

Reply:
This was the reason why we decided on doing that array of parameters in the main abstractquery, and then using `parameterIndex` to signal what parameter to use. That way, we should be able to inject the correct `$<index>` immediately, instead of having to go with a `?` first and replacing it

Reply:
alright, let me rework this

Reply:
I wasn't sure if you decided on doing it differently while I wasn't there, so I suggested something that at least removes the string manipulation 😅 Doing it when converting `AbstractQuery` to the SQL AST would certainly be ideal.

Reply:
I moved the logic to create parameters into the abstract SQL package now. The pg driver can now pass those parameters directly on to the underlying driver and just adds a `$` sign before the indices. Have a look :)